File: bidly-backend/src/main/java/com/bidly/bidly_backend/model/SubastaEstadoAdmin.java
package com.bidly.bidly_backend.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "subasta_estado_admin")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SubastaEstadoAdmin {

    @Id
    @Column(name = "subasta")
    private Long subasta;

    private String estado;

    @Column(name = "alguna_vez_abierta")
    private Boolean algunaVezAbierta = false;

    @Column(name = "fecha_apertura")
    private LocalDateTime fechaApertura;

    @Column(name = "estado_subasta", nullable = false)
    private String estadoSubasta = "pendiente";

    @Column(name = "fecha_inicio_real")
    private LocalDateTime fechaInicioReal;

    @Column(name = "fecha_finalizacion")
    private LocalDateTime fechaFinalizacion;
}

File: bidly-fastapi/app/models/subasta_estado_admin.py
from sqlalchemy import Column, Integer, String, Boolean, DateTime, ForeignKey
from app.database import Base


class SubastaEstadoAdmin(Base):
    __tablename__ = "subasta_estado_admin"

    subasta            = Column(Integer, ForeignKey("subastas.identificador"), primary_key=True)
    estado             = Column(String)
    alguna_vez_abierta = Column(Boolean, default=False)
    fecha_apertura     = Column(DateTime)
    estado_subasta     = Column(String, default="pendiente")
    fecha_inicio_real  = Column(DateTime)
    fecha_finalizacion = Column(DateTime)
